src/main/java/com/commercetools/sync/shoppinglists/helpers/ShoppingListReferenceResolver.java

```diff
@@ -144,4 +152,33 @@ public ShoppingListReferenceResolver(@Nonnull final ShoppingListSyncOptions shop
 
         return completedFuture(draftBuilder);
     }
+
+    /**
+     * Calls the {@code cacheKeysToIds} service methods to fetch all the referenced keys
+     * (i.e custom type, customer) from the CTP to populate caches for the reference resolution.
+     *
+     * <p>Note: This method is meant be only used internally by the library to improve performance.
+     *
+     * @param referencedKeys a wrapper for the custom type and customer references to fetch and cache the id's for.
```

Comment:
did understand from here -> `to fetch and cache the id's for.`

Reply:
Is it better now?
https://github.com/commercetools/commercetools-sync-java/pull/615/commits/939a5993f882d9d508c16ce80c272603db27d062

Reply:
LGTM